### PR TITLE
Adding storage account logging features

### PR DESCRIPTION
**Pull request by Rohit1509**
### Description

Adding diagnostic setting features for Storage account blobs, tables, queues

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List

- [ ] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_storage_account.rb [hunk 1]
```diff
@@ -143,6 +145,144 @@ def table_properties
...
+      {
+        property_name: 'diagnostic_settings',
+        property_endpoint: "#{id}/queueServices/default/providers/microsoft.insights/diagnosticSettings",
+        api_version: @opts[:diagnostic_settings_api_version],
+      },
+    )
+  end
+
+  def has_blobs_read_log_enabled?
+    return false if blobs_diagnostic_settings.nil? || blobs_diagnostic_settings.empty?
```
Comment (sathish-progress) [inline libraries/azure_storage_account.rb:188]: inspec now comes shipped with `activesupport`, so we can use 

```suggestion
    return false if blobs_diagnostic_settings.blank?
```

Comment (Rohit1509) [reply libraries/azure_storage_account.rb]: @sathish-progress , thanks for the review. Made the necessary changes

File: libraries/azure_storage_account.rb [hunk 2]
```diff
@@ -143,6 +145,144 @@ def table_properties
...
+  def has_blobs_read_log_enabled?
+    return false if blobs_diagnostic_settings.blank?
+    result = []
+    blobs_diagnostic_settings.each do |setting|
+      logs = setting.properties&.logs
+      next unless logs
+      result += logs.map { |log| log.enabled if log.category == 'StorageRead' }.compact
+    end
+    result.include?(true)
+  end
```
Comment (sathish-progress) [inline libraries/azure_storage_account.rb:196]: Following DRY principals, we can refactor as below, 


```ruby
def has_blobs_read_log_enabled?
  check_enablement_from(settings: blobs_diagnostic_settings, category: 'StorageRead')
end

def has_tables_read_log_enabled?
  check_enablement_from(settings: tables_diagnostic_settings, category: 'StorageRead')
end

def has_queues_read_log_enabled?
  check_enablement_from(settings: queues_diagnostic_settings, category: 'StorageRead')
end


private 
def check_enablement_from(settings:, category:)
  return false if settings.blank?

  settings.any? do |setting|
    logs = setting.properties&.logs
    logs&.any? { |log| (log.category == category) && log.enabled }
  end
end
```

File: libraries/azure_storage_account.rb [hunk 3]
```diff
@@ -143,6 +145,144 @@ def table_properties
...
+
+  def has_blobs_read_log_enabled?
+    return false if blobs_diagnostic_settings.blank?
+    result = []
+    blobs_diagnostic_settings.each do |setting|
+      logs = setting.properties&.logs
+      next unless logs
+      result += logs.map { |log| log.enabled if log.category == 'StorageRead' }.compact
+    end
```
Comment (sathish-progress) [inline libraries/azure_storage_account.rb:194]: logic can be optimized as below

```ruby 
settings.any? do |setting|
    logs = setting.properties&.logs
    logs&.any? { |log| (log.category == category) && log.enabled }
  end
```

Comment (Rohit1509) [reply libraries/azure_storage_account.rb]: @sathish-progress , this has been implemented

File: libraries/azure_storage_account.rb (hunk 2 above)
Comment (Rohit1509) [reply libraries/azure_storage_account.rb]: @sathish-progress , this has been implemented

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**